### PR TITLE
Update user stories

### LINKED ISSUE #93: Update user stories
Once we have decided on our final set of features based on user stories in our meeting, it may be applicable to make any necessary updates to the user stories in the DG. We may categorise them based on whether they were implemented or not.

We may also consider whether any should be documented in the UG / DG category as "coming in a future version", as this may save us during grading. (Guidelines for determining bugs @ https://nus-cs2103-ay2324s1.github.io/website/schedule/week10/project.html.)

### LINKED ISSUE #130: Update DG
This issue tracks the updating of the DG to match our v1.3 product.

See also #127.

### LINKED ISSUE #159: [PE-D][Tester A] Lack of examples (Delete)
For your delete command, under format, you stated that if duplicate indices are entered, only one would be counted. I feel as though as this is a unique case, it should be highlighted under examples as well for greater clarity.

Do note that this is separate from the previous issue as this focuses on the omission of certain examples rather than outcomes.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: bhnuka/ped#4

### PR DESCRIPTION
**Pull request by mamayuan**
Removed unused user stories in DG according to our previous discussion. 
Added a few more user stories based on our product.
Quick fix to delete example in UG.
Possible to include more user stories. Let me know if yall can think of any. I'll add in :)

Closes #93, closes #159.

### CODE REVIEW

**Comment by mamayuan** [comment]
I'm sorry guys, I forgot to create a new branch for code refactoring. So they are also included in this PR... Not many changes, hopefully it is not too confusing to include two tasks in one here...

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -222,33 +222,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* *`    | user                            | clear all contacts data                                                              | quickly erase all information stored when I will no longer use the app          |
 | `*`      | user who prefers CLI            | use keyboard shortcuts                                                               | perform tasks more efficiently                                                  |
 | `*`      | user                            | see a different background colour for each contact                                   | differentiate between contacts more easily                                      |
-| `*`      | power user with many contacts   | use some scripting capabilities                                                      | automate tedious actions or achieve the exact results I specify                 |
-| `*`      | busy user                       | use icons to denote certain contact information                                      | identify the information I want at a glance                                     |
```
Comment (Cloud7050) [inline docs/DeveloperGuide.md:247]: Is it ok to keep this user story?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -222,33 +222,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-| `*`      | user                            | see a log of past interactions with a contact                                        | know my history with my contacts                                                |
 | `*`      | advanced user                   | search/filter by specific parts of contacts (e.g. containing certain words)          | narrow down contacts to exactly what I am looking for                           |
 | `*`      | user who prefers CLI            | switch between previously entered commands in history                                | easily repeat previous commands                                                 |
-| `*`      | infrequent user                 | view a “cheatsheet” or help dialog for the text commands                             | remember some basic commands I may have forgotten                               |
```
Comment (Cloud7050) [inline docs/DeveloperGuide.md:251]: Similarly, is it ok to keep this?

See also https://github.com/AY2324S1-CS2103-W14-3/tp/issues/171#issuecomment-1793260253.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -159,6 +159,9 @@ Please refer to the [above](#editing-a-contact-edit).
 - `delete 1`
 (Deletes the contact at index #1.)
 
+- `delete 1 1`
+(Deletes the contact at index #1 since duplicate indices count once.)
+
```
Comment (Cloud7050) [inline docs/UserGuide.md:164]: I was initially thinking this would fit better as an "e.g." kind of line under L153.

image

For your consideration.

**Comment by Cloud7050** [comment]
> I'm sorry guys, I forgot to create a new branch for code refactoring. So they are also included in this PR... Not many changes, hopefully it is not too confusing to include two tasks in one here...

Ah I see. It should be possible to cherry pick the commits onto a new branch and then reset this branch to a few commits back, if you want to look into that.

**Comment by Cloud7050** [comment]
Did you mean to autoclose #93 instead of #130? I think the latter is still ongoing. I will edit the body for now.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -236,6 +236,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 | `* *`    | user                            | clear all contacts data                                                              | quickly erase all information stored when I will no longer use the app          |
 | `*`      | user who prefers CLI            | use keyboard shortcuts                                                               | perform tasks more efficiently                                                  |
 | `*`      | user                            | see a different background colour for each contact                                   | differentiate between contacts more easily                                      |
+| `*`      | infrequent user                 | view a “cheatsheet” or help dialog for the text commands                             | remember some basic commands I may have forgotten
```
Comment (Cloud7050) [inline docs/DeveloperGuide.md:239]: I noticed the inverted commas here are "smart quotes" that programs like Word automatically replace in. Should we change these to normal quotes?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -150,18 +150,15 @@ Please refer to the [above](#editing-a-contact-edit).
 
 - You can delete multiple contacts at once by specifying multiple indices separated by spaces.
 
-- Duplicate indices are only counted once.
-
+- Duplicate indices are only counted once.\
+e.g. `delete 1 1` will only deletes the contact at index #`1`.
```
Comment (Cloud7050) [inline docs/UserGuide.md:154]: There is a typo in "deletes".

**Review by Cloud7050 — CHANGES_REQUESTED** [review]

**Comment by mamayuan** [comment]
> Did you mean to autoclose #93 instead of #130? I think the latter is still ongoing. I will edit the body for now.

Yup I was wrong

File: docs/UserGuide.md (hunk 5 above)
Comment (mamayuan) [reply docs/UserGuide.md:154]: Thx! I've fixed it!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (mamayuan) [reply docs/DeveloperGuide.md:239]: ok done!

**Comment by mamayuan** [comment]
> > I'm sorry guys, I forgot to create a new branch for code refactoring. So they are also included in this PR... Not many changes, hopefully it is not too confusing to include two tasks in one here...
> 
> Ah I see. It should be possible to cherry pick the commits onto a new branch and then reset this branch to a few commits back, if you want to look into that.

A bit tedious... I'll leave it like this first. The refactoring is not much.

**Comment by Cloud7050** [comment]
> A bit tedious... I'll leave it like this first. The refactoring is not much.

Aight.

---

I have not looked at the refactoring. But I think the docs part LGTM.

Neutral on removing the icons user story, I guess even if we remove it, we still have enough user stories. But adding icons (as a general idea) may still be a nice future feature.

**Comment by Cloud7050** [comment]
Also works towards #184.

**Comment by mamayuan** [comment]
> | `*` | busy user | use icons to denote certain contact information | identify the information I want at a glance |



> > A bit tedious... I'll leave it like this first. The refactoring is not much.
> 
> Aight.
> 
> I have not looked at the refactoring. But I think the docs part LGTM.
> 
> Neutral on removing the icons user story, I guess even if we remove it, we still have enough user stories. But adding icons (as a general idea) may still be a nice future feature.

Yup added it back

**Review by Cloud7050 — APPROVED** [review]

**Review by bwangpj — APPROVED** [review]

**Pull request merged**